Realm/RLMObjectSchema.mm

```diff
@@ -188,6 +199,24 @@ + (NSArray *)propertiesForClass:(Class)objectClass isSwift:(bool)isSwiftClass {
         }
     }
 
+    if (NSArray *optionalProperties = [objectUtil getOptionalPropertyNames:swiftObjectInstance]) {
```

Comment:
Except we're already running code in a `isSwiftClass` conditional above, and you could use an `else` statement for the `requiredPropertiesForClass:` just after. You could avoid the stub implementations of `getOptionalPropertyNames` in ObjC and `requiredPropertiesForClass` in Swift.